### ISSUE TITLE
S1172 Unused parameters for method syntax defined by attributes

### ISSUE THREAD

**Issue by njannink** [SonarSource-VisualStudio/sonarlint-roslyn-analyzer#413]
Gonna be difficult to detect, but for example the following code examples gives a false positive on the context parameter. But it cannot be removed, because the Attribute on top of the method requires the argument to be in the method syntax

```
public class MyTestClass
{
    private static object obj;

    [ClassInitialize]
    public static void ClassInitialize(TestContext context) // S1172 on 'context' parameter
    {
        obj = new object();
    }

    [TestMethod]
    public void Test1()
    {
        Assert.IsNotNull(obj);
    }
}

public class MyDto
{
    public string Name { get; set; }

    [OnDeserialized]
    private void OnDeserialized(StreamingContext context)
    {
        
    }
}
```

**Comment by tvsonar** [comment]
Thanks @njannink for reporting this. I'm not sure if we can come up with a full list of attributes that are similar. We usually try not to maintain lists of exceptions in the rules. 

Do you know other widely used attributes that require method parameters?

**Comment by njannink** [comment]
These are the ones I know, but there are probably many more. You can actually write an opposite rule also that fails when methods marked with this attribute are lacking the required arguments.

```
System.Runtime.Serialization.OnSerializingAttribute
System.Runtime.Serialization.OnSerializedAttribute
System.Runtime.Serialization.OnDeserializingAttribute
System.Runtime.Serialization.OnDeserializedAttribute

Microsoft.VisualStudio.TestTools.UnitTesting.AssemblyInitializeAttribute
Microsoft.VisualStudio.TestTools.UnitTesting.ClassInitializeAttribute
```

**Comment by tvsonar** [comment]
Thanks @njannink for the list.
